tests/test_approximate.py

```diff
@@ -0,0 +1,489 @@
+                                 neighborhood_policy=NeighborhoodPolicy.LSHNearest(n_dimensions=2),
+                                 context_history=[[0, 1, 2, 3, 5], [1, 1, 1, 1, 1], [0, 0, 1, 0, 0],
+                                                  [0, 2, 2, 3, 5], [1, 3, 1, 1, 1], [0, 0, 0, 0, 0],
+                                                  [0, 1, 4, 3, 5], [0, 1, 2, 4, 5], [1, 2, 1, 1, 3],
+                                                  [0, 2, 1, 0, 0]],
+                                 contexts=[[0, 1, 2, 3, 5], [1, 1, 1, 1, 1]],
+                                 seed=123456,
+                                 num_run=1,
+                                 is_predict=True)
+
+        self.assertListEqual(arms, [1, 3])
+
+    def test_no_neighbors_hash(self):
+        contexts = [[0, -1, -2, -3, -5], [-1, -1, -1, -1, -1], [0, -1, -2, -3, -5], [-1, -1, -1, -1, -1],
+                                           [0, -1, -2, -3, -5]]
+        arms, mab = self.predict(arms=[1, 2, 3, 4],
+                                 decisions=[1, 1, 1, 2, 2, 2],
+                                 rewards=[10, 10, 10, -10, -10, -10],
+                                 learning_policy=LearningPolicy.EpsilonGreedy(epsilon=0),
+                                 neighborhood_policy=NeighborhoodPolicy.LSHNearest(n_dimensions=20, n_tables=1),
+                                 context_history=[[1, 1, 2, 3, 5], [1, 2, 1, 1, 1], [0, 0, 1, 0, 0],
+                                                  [0, 2, 2, 3, 5], [1, 3, 1, 1, 1], [0, 0, 0, 0, 0]],
+                                 contexts=contexts,
+                                 seed=7,
+                                 num_run=1,
+                                 is_predict=True)
+
+        hashes = mab._imp.get_context_hash(np.asarray(contexts), mab._imp.table_to_plane[0])
+        for h in hashes:
+            self.assertEqual(len(mab._imp.table_to_hash_to_index[0][h]), 0)
+
+        arms, mab = self.predict(arms=[1, 2, 3, 4],
+                                 decisions=[1, 1, 1, 2, 2, 2],
+                                 rewards=[10, 10, 10, -10, -10, -10],
+                                 learning_policy=LearningPolicy.EpsilonGreedy(epsilon=0),
+                                 neighborhood_policy=NeighborhoodPolicy.LSHNearest(n_dimensions=20, n_tables=1),
+                                 context_history=[[1, 1, 2, 3, 5], [1, 2, 1, 1, 1], [0, 0, 1, 0, 0],
+                                                  [0, 2, 2, 3, 5], [1, 3, 1, 1, 1], [0, 0, 0, 0, 0]],
+                                 contexts=contexts,
+                                 seed=12,
+                                 num_run=1,
+                                 is_predict=True)
+
+        hashes = mab._imp.get_context_hash(np.asarray(contexts), mab._imp.table_to_plane[0])
+        for h in hashes:
+            self.assertEqual(len(mab._imp.table_to_hash_to_index[0][h]), 0)
```

Comment:
I like the idea of checking the internals of hashes for behavior. 

This seems to be the case for the case when there are no nhodds. 

Is there a similar internal checking test for the case where there are neighbors where they should be assigned together etc?

Reply:
Added